src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -759,6 +760,34 @@ PurgeConnectionForPlacement(ShardPlacement *placement)
 
 
 /*
+ * Simply removes a given connection from the transaction participant hash.
+ */
+static void
+RemoveXactConnection(PGconn *connection)
```

Comment:
Added a comment to explain why I didn't want to reuse the code (extracting host and port from the connection is unnecessary/expensive (traversal of connection options in a list) when we already have a `ShardPlacement`, so I reserve that method for when all we have handy is a `connection`).